### PR TITLE
Extra set cmdID logic to method

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: rpc/send_test.go [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
-	// The provided addrs is nil, so its length will be always
-	// less than the specified response number
-	ret, err := Send(Options{N: 1}, "", nil, nil, nil, nil)
-
-	// the expected return is nil and SendError
-	if _, ok := err.(SendError); !ok || ret != nil {
-		t.Fatalf("Shorter addrs should return nil and SendError.")
-	}
-}
+// Copyright 2015 The Cockroach Authors.
```
Comment (tschottdorf) [inline rpc/send_test.go]: is this a line ending change for the better?

**Comment by tschottdorf** [comment]
LGTM

**Comment by tschottdorf** [comment]
LGTM

**Comment by petermattis** [comment]
LGTM

File: rpc/send_test.go (hunk 1 above)
Comment (ngaut) [inline rpc/send_test.go]: You mean it does not need to format ? I am confused. It's formatted by command "go fmt".

Comment (bdarnell) [inline rpc/send_test.go]: This file had DOS-style line endings (CRLF) before; your change fixed it to have unix-style line endings (LF). That's what we want, but github gets confused by this and shows the entire file as being rewritten.

File: storage/range.go [hunk 2]
```diff
@@ -437,6 +437,21 @@ func (r *Range) addReadOnlyCmd(method string, args proto.Request, reply proto.Re
 	return err
 }
 
+// getCmdID will create a ClientCmdId if it's empty in Request, otherwise
+// just return it.
```
Comment (spencerkimball) [inline storage/range.go]: s/otherwise just return it/otherwise, returns existing value./

**Comment by spencerkimball** [comment]
LGTM

**Pull request merged**